### PR TITLE
Refactored price functions and added more tests

### PR DESCRIPTION
**Pull request by monstrobishi**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

Refactored price functions and added more tests

#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

File: packages/salmon-oracles-functions/__tests__/test.utils.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import { Bech32, Elliptic, EllipticPair, dSHA256 } from '@defichain/jellyfish-crypto'
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { CTransaction, CTransactionSegWit, Transaction, TransactionSegWit } from '@defichain/jellyfish-transaction'
+import { SmartBuffer } from 'smart-buffer'
```
Comment (fuxingloh) [inline packages/salmon-oracles-functions/__tests__/test.utils.ts:4]: ignore coverage https://github.com/DeFiCh/jellyfish/blob/26068ca73906f739bd0e196e4493d27568c93e15/packages/jellyfish-transaction-builder/jest.config.js#L12-L15

**Review by jingyi2811 — APPROVED** [review]

File: packages/salmon-oracles-functions/__tests__/test.utils.ts (hunk 1 above)
Comment (fuxingloh) [reply packages/salmon-oracles-functions/__tests__/test.utils.ts:4]: Realized you are using module mapper, it should be here instead https://github.com/DeFiCh/salmon/blob/main/jest.config.js

Comment (monstrobishi) [reply packages/salmon-oracles-functions/__tests__/test.utils.ts:4]: Should I delete the `jest.config.js` inside the module

Comment (monstrobishi) [reply packages/salmon-oracles-functions/__tests__/test.utils.ts:4]: Ok looking at `whale` the answer seems to be yes

**Review by fuxingloh — CHANGES_REQUESTED** [review]
should separate out into each package, easier to things merged in. instead of working on one large PR.

Kinda like shift left what is easy to merge, and for those that require long thoughtful discussion. Especially for maintainability.

File: packages/salmon-price-functions/src/index.ts [hunk 2]
```diff
@@ -1,49 +1,66 @@
...
+  for (let i = 0, path = keyPath.split('.'), len = path.length; i < len; i++) {
+    obj = obj[path[i]]
+  };
+  return obj
+}
 export class PriceManager {
-  async fetchValueFromSource (url: string, jsonPath: string, apiToken?: string): Promise<BigNumber> {
-    const response = await fetch(`${url}&token=${apiToken ?? ''}`, {
+  private async fetchMultipleAssets (config: PriceSourceConfig): Promise<AssetPrice[]> {
+    const url = config.baseUrl
```
[the remark this answers is not in the document]
Comment (fuxingloh) [reply packages/salmon-price-functions/src/index.ts:31]: - Some might even offer swagger
- Or we might even read from another blockchain. 
- Or using other protocols that are not URL-based. 

Provider pattern is easy and more agnostic in the long run.

File: packages/salmon-price-functions/package-lock.json [hunk 3]
```diff
@@ -0,0 +1,1383 @@
...
+  "requires": true,
+  "packages": {
+    "": {
+      "name": "@defichain/salmon-price-functions",
+      "version": "0.0.0",
+      "license": "MIT",
+      "dependencies": {
+        "@defichain/jellyfish-crypto": "next",
+        "@defichain/jellyfish-transaction": "next",
+        "@defichain/jellyfish-transaction-builder": "next",
```
Comment (fuxingloh) [inline packages/salmon-price-functions/package-lock.json:14]: this file should never exist in mono repo setup. Likely your setup is wrong

Comment (fuxingloh) [reply packages/salmon-price-functions/package-lock.json:14]: There should only be one package lock at project root.

**Review by fuxingloh — CHANGES_REQUESTED** [review]

**Comment by monstrobishi** [comment]
Closing to split into multiple PRs with above feedback in mind

**Comment by monstrobishi** [comment]
Keeping branch open for now for reference

**Pull request closed without merging**